🐛 [BUG] `apx dev check` fails when route tree is not generated

When running `apx dev check` in a project without a UI build, TypeScript compilation fails because the TanStack Router route tree file has not been generated.

## To Reproduce

1. Create a new apx project or work with an existing one
2. Run `uv run apx dev check` before building the UI or generating the route tree
3. Observe the TypeScript compilation error

## Error Output

```
❌ [tsc] TypeScript compilation failed
src/jview/ui/main.tsx(5,27): error TS2307: Cannot find module '@/types/routeTree.gen' or its corresponding type declarations.
src/jview/ui/routes/_sidebar/profile.tsx(29,38): error TS2345: Argument of type '"/_sidebar/profile"' is not assignable to parameter of type 'undefined'.
src/jview/ui/routes/_sidebar/route.tsx(12,38): error TS2345: Argument of type '"/_sidebar"' is not assignable to parameter of type 'undefined'.
src/jview/ui/routes/index.tsx(7,38): error TS2345: Argument of type '"/"' is not assignable to parameter of type 'undefined'.

error: "tsc" exited with code 1
```

## Root Cause

The TanStack Router route tree file (`@/types/routeTree.gen`) has not been generated, causing TypeScript to fail when checking type definitions.

## Expected Behavior

`apx dev check` should either:
- Automatically generate the route tree before running type checks
- Gracefully handle the missing file with a helpful error message
- Skip the route tree type check if the file doesn't exist yet

## Workaround

Build the UI first to generate the route tree:
```bash
uv run apx ui build
uv run apx dev check
```

--- a/src/apx/assets/entrypoint.ts
+++ b/src/apx/assets/entrypoint.ts
@@ -2,6 +2,7 @@ import { build, createServer, type InlineConfig, type Plugin } from "vite";
 import tailwindcss from "@tailwindcss/vite";
 import react from "@vitejs/plugin-react";
 import { tanstackRouter } from "@tanstack/router-plugin/vite";
+import { Generator, getConfig } from "@tanstack/router-generator";
 import type { IncomingMessage, ServerResponse } from "http";
 import {
   LoggerProvider,
@@ -362,6 +363,18 @@ async function runBuild() {
   await build(config);
 }
 
+async function runGenerate() {
+  const config = getConfig({
+    target: "react",
+    autoCodeSplitting: true,
+    routesDirectory: `${uiRoot}/routes`,
+    generatedRouteTree: `${uiRoot}/types/routeTree.gen.ts`,
+  });
+
+  const generator = new Generator({ config, root: uiRoot });
+  await generator.run();
+}
+
 // Main entry point
 if (mode === "dev") {
   runDev().catch((err) => {
@@ -373,7 +386,12 @@ if (mode === "dev") {
     logError(`Failed to build: ${err}`);
     process.exit(1);
   });
+} else if (mode === "generate") {
+  runGenerate().catch((err) => {
+    logError(`Failed to generate route tree: ${err}`);
+    process.exit(1);
+  });
 } else {
-  logError(`Invalid mode: ${mode}. Expected "dev" or "build".`);
+  logError(`Invalid mode: ${mode}. Expected "dev", "build", or "generate".`);
   process.exit(1);
 }

--- a/src/cli/dev/check.rs
+++ b/src/cli/dev/check.rs
@@ -1,8 +1,9 @@
 use clap::Args;
 use std::path::PathBuf;
 
+use crate::cli::frontend::common::prepare_frontend_args;
 use crate::cli::run_cli_async;
-use crate::common::{BunCommand, run_preflight_checks};
+use crate::common::{BunCommand, ensure_entrypoint_deps, run_preflight_checks};
 use tokio::process::Command as TokioCommand;
 use tracing::debug;
 
@@ -26,6 +27,9 @@ pub async fn run_inner(args: CheckArgs) -> Result<(), String> {
 
     // Run preflight checks (installs deps if needed)
     run_preflight_checks(&app_dir).await?;
+
+    // Generate route tree (must complete before tsc)
+    generate_route_tree(&app_dir).await?;
 
     // Run tsc -b --incremental in one tokio thread
     let bun = BunCommand::new()?;
@@ -145,5 +149,37 @@ pub async fn run_inner(args: CheckArgs) -> Result<(), String> {
     }
 
     // If no errors, just move forward
+    Ok(())
+}
+
+async fn generate_route_tree(app_dir: &std::path::Path) -> Result<(), String> {
+    println!("Generating route tree...");
+
+    // Ensure entrypoint deps are installed (includes @tanstack/router-generator)
+    ensure_entrypoint_deps(app_dir).await?;
+
+    // Prepare frontend args for "generate" mode
+    let (entrypoint, args, app_name) = prepare_frontend_args(app_dir, "generate")?;
+
+    // Run: bun run <entrypoint.ts> generate <uiRoot> <outDir> <publicDir>
+    let bun = BunCommand::new()?;
+    let output = bun
+        .tokio_command()
+        .arg("run")
+        .arg(&entrypoint)
+        .args(&args)
+        .env("APX_APP_NAME", &app_name)
+        .current_dir(app_dir)
+        .output()
+        .await
+        .map_err(|err| format!("Failed to run route tree generation: {err}"))?;
+
+    if !output.status.success() {
+        let stdout = String::from_utf8_lossy(&output.stdout);
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        return Err(format!("Route tree generation failed:\n{stdout}\n{stderr}"));
+    }
+
+    println!("Route tree generated");
     Ok(())
 }

--- a/src/cli/frontend/mod.rs
+++ b/src/cli/frontend/mod.rs
@@ -1,3 +1,3 @@
 pub mod build;
-mod common;
+pub(crate) mod common;
 pub mod dev;

--- a/src/common.rs
+++ b/src/common.rs
@@ -18,6 +18,7 @@ pub const ENTRYPOINT_DEV_DEPS: &[&str] = &[
     "@tailwindcss/vite",
     "@vitejs/plugin-react",
     "@tanstack/router-plugin",
+    "@tanstack/router-generator",
     "@opentelemetry/sdk-logs",
     "@opentelemetry/exporter-logs-otlp-http",
     "@opentelemetry/api-logs",

